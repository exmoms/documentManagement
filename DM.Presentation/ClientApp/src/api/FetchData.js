/**
 * @file this file includes functions that fetch data from API
 */

// Generic Delete Method
export async function deleteFromApi(url) {
  let res = await fetch(url, {
    method: "DELETE",
    headers: {
      Accept: "application/json",
      "Content-Type": "application/json",
    },
  });
  return res;
}

// Get MetaData Models
export async function fetchData(url) {
  let object = {};
  /*
   * send the request through a CORS proxy
   * https://stackoverflow.com/questions/43262121/trying-to-use-fetch-and-pass-in-mode-no-cors/43268098#43268098
   */
  await fetch(url, {
    method: "GET",
    headers: {
      Accept: "application/json",
      "Content-Type": "application/json",
    },
  })
    .then((response) => response.json())
    .then(
      (data) => (object = data),
      (error) => console.log(error)
    );
  return object;
}
