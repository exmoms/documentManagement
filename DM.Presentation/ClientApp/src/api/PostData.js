/**
 * @file This file includes functions that help to post data to a specific API
 * @version 1.0
 */

/**
 * Post new document data.
 * Upload form-data, scanned copy of doc and attachments.
 * @author Youssef Shaaban <y.shaaban@lit-co.net>
 * @param {*} documentData
 * @param {*} images
 * @param {*} attachments
 * @param {*} url
 * @param {*} option its values either "Add" for add new document version or "Update" to update exsiting document Version
 */
export async function postDocumentData(
  documentData,
  scannedPages,
  attachments,
  url = "",
  option
) {
  var formData = new FormData();

  if (process.env.NODE_ENV === "development") {
    console.log(documentData);
    console.log(attachments);
    console.log(scannedPages);
  }

  switch (option) {
    case "Add":
      formData.append("Document", JSON.stringify(documentData));
      break;
    case "Update":
      formData.append("Version", JSON.stringify(documentData));
      break;
    default:
  }

  attachments.forEach((attachment) => {
    formData.append("Attachments", attachment);
  });
  scannedPages.forEach((scannedPage) => {
    formData.append("Scans", scannedPage);
  });

  let response = {};
  await fetch(url, {
    // content-type header should not be specified!
    method: "POST",
    headers: {
      Accept: "application/json",
      // "Content-Type": "application/json",
    },
    body: formData,
  }).then(
    (res) => {
      response = res;
      // response.json();
    },
    (error) => console.log("[ERROR]:", error)
  );
  return response;
}

/**
 * Post Attachment Data.
 * Upload attachment json data + attachment file  .
 * @author Ali Daghman <ali.daghman@lit-co.net>
 * @param {*} payload
 * @param {*} attachmentFile
 * @param {*} url
 */
export async function postAttachmentData(payload, attachmentFile, url = "") {
  var formData = new FormData();

  formData.append("Attachment", JSON.stringify(payload));
  formData.append("FileAttachment", attachmentFile);

  let response = {};
  await fetch(url, {
    // content-type header should not be specified!
    method: "POST",
    body: formData,
  }).then(
    (res) => {
      response = res;
      // response.json();
    },
    (error) => console.log("[ERROR]:", error)
  );
  return response;
}

/**
 * Post Request To API
 * Send post request to api with  payload  and return a response.
 * @author Ali Daghman <ali.daghman@lit-co.net>
 * @param {*} url The api url
 * @param {*} payload js object which will be sent as request body.
 */
export async function postDataToAPI(url, payload = null, method = "POST") {

  const response = await fetch(url, {
    method: method,
    headers: {
      Accept: "application/json",
      "Content-Type": "application/json",
    },
    body: JSON.stringify(payload),
  });
  return response;
}
