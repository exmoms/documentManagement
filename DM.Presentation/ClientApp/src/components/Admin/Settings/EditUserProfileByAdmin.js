import { FormHelperText } from "@material-ui/core";
import Container from "@material-ui/core/Container";
import CssBaseline from "@material-ui/core/CssBaseline";
import FormControl from "@material-ui/core/FormControl";
import Grid from "@material-ui/core/Grid";
import IconButton from "@material-ui/core/IconButton";
import InputAdornment from "@material-ui/core/InputAdornment";
import InputLabel from "@material-ui/core/InputLabel";
import OutlinedInput from "@material-ui/core/OutlinedInput";
import { makeStyles } from "@material-ui/core/styles";
import TextField from "@material-ui/core/TextField";
import Visibility from "@material-ui/icons/Visibility";
import VisibilityOff from "@material-ui/icons/VisibilityOff";
import clsx from "clsx";
import React, { useEffect, useState } from "react";
// import { postData } from "../../api/PostData";
import { useTranslation } from "react-i18next";

const useStyles = makeStyles((theme) => ({
  paper: {
    marginTop: theme.spacing(8),
    display: "flex",
    flexDirection: "column",
    alignItems: "center",
  },
  form: {
    width: "100%", // Fix IE 11 issue.
    marginTop: theme.spacing(3),
  },
  submit: {
    margin: theme.spacing(3, 0, 2),
  },
  margin: {
    margin: theme.spacing(1),
  },
  textField: {
    width: "100%",
  },
}));

export default function EditUserProfileByAdmin(props) {
  const { t } = useTranslation();
  const classes = useStyles();

  const [email, setemail] = useState("");
  const [tel, settel] = useState("");
  const [values, setValues] = useState({
    password: "",
    showPassword: false,
  });

  useEffect(() => {
    if (props.profileData !== null) {
      setemail(props.profileData.email);
      settel(
        props.profileData.phoneNumber === null
          ? ""
          : props.profileData.phoneNumber
      );
    }
  }, [props]);

  const onChange = (event) => {
    if (event.target.name === "edit-profile-by-admin-email") {
      setemail(event.target.value);
    }
    if (event.target.name === "edit-profile-by-admin-tel") {
      settel(event.target.value);
    }
  };

  const handleChange = (prop) => (event) => {
    setValues({ ...values, [prop]: event.target.value });
  };

  const handleClickShowPassword = () => {
    setValues({ ...values, showPassword: !values.showPassword });
  };

  const handleMouseDownPassword = (event) => {
    event.preventDefault();
  };

  return (
    <Container component="main" maxWidth="xs">
      <CssBaseline />
      <div className={classes.paper}>
        <Grid container spacing={1}>
          <Grid item xs={12}>
            <TextField
              variant="outlined"
              required
              fullWidth
              value={email}
              onChange={onChange}
              name="edit-profile-by-admin-email"
              label={t("email")}
              type="email"
              id="edit-profile-by-admin-email"
              autoComplete="email"
            />
          </Grid>
          <Grid item xs={12}>
            <TextField
              variant="outlined"
              fullWidth
              value={tel}
              onChange={onChange}
              name="edit-profile-by-admin-tel"
              label={t("phone_number")}
              type="tel"
              id="edit-profile-by-admin-tel"
              autoComplete="tel"
            />
          </Grid>
          <FormControl
            className={clsx(classes.margin, classes.textField)}
            variant="outlined"
          >
            <InputLabel
              error={!!props.errors.newPasswordErrors}
              htmlFor="edit-profile-by-admin-outlined-adornment-password"
            >
              {t("new_password")}
            </InputLabel>
            <OutlinedInput
              error={!!props.errors.newPasswordErrors}
              id="edit-profile-by-admin-outlined-adornment-password"
              type={values.showPassword ? "text" : "password"}
              value={values.password}
              onChange={handleChange("password")}
              endAdornment={
                <InputAdornment position="end">
                  <IconButton
                  id="iconButton"
                    aria-label="toggle password visibility"
                    onClick={handleClickShowPassword}
                    onMouseDown={handleMouseDownPassword}
                    edge="end"
                  >
                    {values.showPassword ? <Visibility /> : <VisibilityOff />}
                  </IconButton>
                </InputAdornment>
              }
              labelWidth={120}
            />
            <FormHelperText
              error={!!props.errors.newPasswordErrors}
              id="edit-profile-by-admin-outlined-adornment-password-helper-text"
            >
              {props.errors.newPasswordErrors}
            </FormHelperText>
          </FormControl>
          <Grid item xs={12}>
            <TextField
              variant="outlined"
              fullWidth
              name="edit-profile-by-admin-confirmPassword"
              label={t("confirmNewPassword_label")}
              type="password"
              error={!!props.errors.confirmErrors}
              helperText={props.errors.confirmErrors}
              id="edit-profile-by-admin-confirmPassword"
              autoComplete="current-password"
            />
          </Grid>
        </Grid>
      </div>
    </Container>
  );
}
