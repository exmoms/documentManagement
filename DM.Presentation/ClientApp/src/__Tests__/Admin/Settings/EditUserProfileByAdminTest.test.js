import React from "react";
import ReactDom from "react-dom";
import EditUserProfileByAdmin from "../../../components/Admin/Settings/EditUserProfileByAdmin";
import { cleanup } from "@testing-library/react";
import { shallow, mount } from "enzyme";
import toJSON from "enzyme-to-json";
import TextField from "@material-ui/core/TextField";
import { act } from "react-dom/test-utils";
import Button from "@material-ui/core/Button";
import { BrowserRouter as Router } from "react-router-dom";
import * as FetchMock from "../../../api/FetchData";
import * as PostDataMock from "../../../api/PostData";
import Typography from "@material-ui/core/Typography";
import Dialog from "@material-ui/core/Dialog";
import Visibility from "@material-ui/icons/Visibility";
import VisibilityOff from "@material-ui/icons/VisibilityOff";
afterEach(cleanup);
describe("EditUserProfileByAdmin", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });
  test("renders without crashing", () => {
    const div = document.createElement("div");
    let profileData={email:"testedValue@gmail.com" , phoneNumber:123456788};
    let errors={newPasswordErrors:"newPasswordErrors" , confirmErrors:"confirmErrors"};
    ReactDom.render(
        <EditUserProfileByAdmin profileData={profileData} errors={errors}/>,
      div
    );
  });

  test("matches snapshot", () => {
    let profileData={email:"testedValue@gmail.com" , phoneNumber:123456788};
    let errors={newPasswordErrors:"newPasswordErrors" , confirmErrors:"confirmErrors"};
    const wrapper = shallow(
        <EditUserProfileByAdmin profileData={profileData} errors={errors}/>
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
 
  test("intialization ProfilePopUpByAdmin", () => {
    let profileData={email:"testedValue@gmail.com" , phoneNumber:123456788};
    let errors={newPasswordErrors:"newPasswordErrors" , confirmErrors:"confirmErrors"};
  let  wrapper = mount(<EditUserProfileByAdmin profileData={profileData} errors={errors} />) ;
  expect(wrapper.find("#edit-profile-by-admin-email").at(0).props().value).toBe("testedValue@gmail.com");
  expect(wrapper.find("#edit-profile-by-admin-tel").at(0).props().value).toBe(123456788);
  expect(wrapper.find("#edit-profile-by-admin-outlined-adornment-password").at(0).props().error).toBeTruthy();
  expect(wrapper.find("#edit-profile-by-admin-outlined-adornment-password").at(0).props().type).toBe("password");
  expect(wrapper.find("#edit-profile-by-admin-outlined-adornment-password").at(0).props().value).toBe("");
  expect(wrapper.find(Visibility).length).toBe(0);
  expect(wrapper.find(VisibilityOff).length).toBe(1);
  expect(wrapper.find("#edit-profile-by-admin-outlined-adornment-password-helper-text").at(0).props().error).toBeTruthy();
  expect(wrapper.find("#edit-profile-by-admin-confirmPassword").at(0).props().error).toBeTruthy();
  expect(wrapper.find("#edit-profile-by-admin-confirmPassword").at(0).props().helperText).toBe("confirmErrors");
});
test("intialization ProfilePopUpByAdmin when phoneNumber==null", () => {
    let profileData={email:"testedValue@gmail.com" , phoneNumber:null};
    let errors={newPasswordErrors:"newPasswordErrors" , confirmErrors:"confirmErrors"};
  let  wrapper = mount(<EditUserProfileByAdmin profileData={profileData} errors={errors} />) ;
  expect(wrapper.find("#edit-profile-by-admin-email").at(0).props().value).toBe("testedValue@gmail.com");
  expect(wrapper.find("#edit-profile-by-admin-tel").at(0).props().value).toBe("");
  expect(wrapper.find("#edit-profile-by-admin-outlined-adornment-password").at(0).props().error).toBeTruthy();
  expect(wrapper.find("#edit-profile-by-admin-outlined-adornment-password").at(0).props().type).toBe("password");
  expect(wrapper.find("#edit-profile-by-admin-outlined-adornment-password").at(0).props().value).toBe("");
  expect(wrapper.find(Visibility).length).toBe(0);
  expect(wrapper.find(VisibilityOff).length).toBe(1);
  expect(wrapper.find("#edit-profile-by-admin-outlined-adornment-password-helper-text").at(0).props().error).toBeTruthy();
  expect(wrapper.find("#edit-profile-by-admin-confirmPassword").at(0).props().error).toBeTruthy();
  expect(wrapper.find("#edit-profile-by-admin-confirmPassword").at(0).props().helperText).toBe("confirmErrors");
});
test("intialization ProfilePopUpByAdmin when profileData==null", () => {
    let profileData=null;
    let errors={newPasswordErrors:"newPasswordErrors" , confirmErrors:"confirmErrors"};
  let  wrapper = mount(<EditUserProfileByAdmin profileData={profileData} errors={errors} />) ;
  expect(wrapper.find("#edit-profile-by-admin-email").at(0).props().value).toBe("");
  expect(wrapper.find("#edit-profile-by-admin-tel").at(0).props().value).toBe("");
  expect(wrapper.find("#edit-profile-by-admin-outlined-adornment-password").at(0).props().error).toBeTruthy();
  expect(wrapper.find("#edit-profile-by-admin-outlined-adornment-password").at(0).props().type).toBe("password");
  expect(wrapper.find("#edit-profile-by-admin-outlined-adornment-password").at(0).props().value).toBe("");
  expect(wrapper.find(Visibility).length).toBe(0);
  expect(wrapper.find(VisibilityOff).length).toBe(1);
  expect(wrapper.find("#edit-profile-by-admin-outlined-adornment-password-helper-text").at(0).props().error).toBeTruthy();
  expect(wrapper.find("#edit-profile-by-admin-confirmPassword").at(0).props().error).toBeTruthy();
  expect(wrapper.find("#edit-profile-by-admin-confirmPassword").at(0).props().helperText).toBe("confirmErrors");
});
test("onChange functionality when event.target.name === edit-profile-by-admin-email", () => {
    let profileData={email:"testedValue@gmail.com" , phoneNumber:123456788};
    let errors={newPasswordErrors:"newPasswordErrors" , confirmErrors:"confirmErrors"};
  let  wrapper = mount(<EditUserProfileByAdmin profileData={profileData} errors={errors} />) ;
  expect(wrapper.find("#edit-profile-by-admin-email").at(0).props().value).toBe("testedValue@gmail.com");
  expect(wrapper.find("#edit-profile-by-admin-tel").at(0).props().value).toBe(123456788);
  expect(wrapper.find("#edit-profile-by-admin-outlined-adornment-password").at(0).props().error).toBeTruthy();
  expect(wrapper.find("#edit-profile-by-admin-outlined-adornment-password").at(0).props().type).toBe("password");
  expect(wrapper.find("#edit-profile-by-admin-outlined-adornment-password").at(0).props().value).toBe("");
  expect(wrapper.find(Visibility).length).toBe(0);
  expect(wrapper.find(VisibilityOff).length).toBe(1);
  expect(wrapper.find("#edit-profile-by-admin-outlined-adornment-password-helper-text").at(0).props().error).toBeTruthy();
  expect(wrapper.find("#edit-profile-by-admin-confirmPassword").at(0).props().error).toBeTruthy();
  expect(wrapper.find("#edit-profile-by-admin-confirmPassword").at(0).props().helperText).toBe("confirmErrors");

  act(() => {
    wrapper
      .find("#edit-profile-by-admin-email")
      .at(0)
      .props()
      .onChange({ target: { value: "testedValue2@gmail.com" ,name:"edit-profile-by-admin-email"} });
  });
  wrapper.render();
    wrapper.update();

    expect(wrapper.find("#edit-profile-by-admin-email").at(0).props().value).toBe("testedValue2@gmail.com");
    expect(wrapper.find("#edit-profile-by-admin-tel").at(0).props().value).toBe(123456788);
    expect(wrapper.find("#edit-profile-by-admin-outlined-adornment-password").at(0).props().error).toBeTruthy();
    expect(wrapper.find("#edit-profile-by-admin-outlined-adornment-password").at(0).props().type).toBe("password");
    expect(wrapper.find("#edit-profile-by-admin-outlined-adornment-password").at(0).props().value).toBe("");
    expect(wrapper.find(Visibility).length).toBe(0);
    expect(wrapper.find(VisibilityOff).length).toBe(1);
    expect(wrapper.find("#edit-profile-by-admin-outlined-adornment-password-helper-text").at(0).props().error).toBeTruthy();
    expect(wrapper.find("#edit-profile-by-admin-confirmPassword").at(0).props().error).toBeTruthy();
    expect(wrapper.find("#edit-profile-by-admin-confirmPassword").at(0).props().helperText).toBe("confirmErrors");
});

test("onChange functionality when event.target.name === edit-profile-by-admin-tel", () => {
    let profileData={email:"testedValue@gmail.com" , phoneNumber:123456788};
    let errors={newPasswordErrors:"newPasswordErrors" , confirmErrors:"confirmErrors"};
  let  wrapper = mount(<EditUserProfileByAdmin profileData={profileData} errors={errors} />) ;
  expect(wrapper.find("#edit-profile-by-admin-email").at(0).props().value).toBe("testedValue@gmail.com");
  expect(wrapper.find("#edit-profile-by-admin-tel").at(0).props().value).toBe(123456788);
  expect(wrapper.find("#edit-profile-by-admin-outlined-adornment-password").at(0).props().error).toBeTruthy();
  expect(wrapper.find("#edit-profile-by-admin-outlined-adornment-password").at(0).props().type).toBe("password");
  expect(wrapper.find("#edit-profile-by-admin-outlined-adornment-password").at(0).props().value).toBe("");
  expect(wrapper.find(Visibility).length).toBe(0);
  expect(wrapper.find(VisibilityOff).length).toBe(1);
  expect(wrapper.find("#edit-profile-by-admin-outlined-adornment-password-helper-text").at(0).props().error).toBeTruthy();
  expect(wrapper.find("#edit-profile-by-admin-confirmPassword").at(0).props().error).toBeTruthy();
  expect(wrapper.find("#edit-profile-by-admin-confirmPassword").at(0).props().helperText).toBe("confirmErrors");

  act(() => {
    wrapper
      .find("#edit-profile-by-admin-email")
      .at(0)
      .props()
      .onChange({ target: { value: "000000077777" ,name:"edit-profile-by-admin-tel"} });
  });
  wrapper.render();
    wrapper.update();

    expect(wrapper.find("#edit-profile-by-admin-email").at(0).props().value).toBe("testedValue@gmail.com");
    expect(wrapper.find("#edit-profile-by-admin-tel").at(0).props().value).toBe("000000077777");
    expect(wrapper.find("#edit-profile-by-admin-outlined-adornment-password").at(0).props().error).toBeTruthy();
    expect(wrapper.find("#edit-profile-by-admin-outlined-adornment-password").at(0).props().type).toBe("password");
    expect(wrapper.find("#edit-profile-by-admin-outlined-adornment-password").at(0).props().value).toBe("");
    expect(wrapper.find(Visibility).length).toBe(0);
    expect(wrapper.find(VisibilityOff).length).toBe(1);
    expect(wrapper.find("#edit-profile-by-admin-outlined-adornment-password-helper-text").at(0).props().error).toBeTruthy();
    expect(wrapper.find("#edit-profile-by-admin-confirmPassword").at(0).props().error).toBeTruthy();
    expect(wrapper.find("#edit-profile-by-admin-confirmPassword").at(0).props().helperText).toBe("confirmErrors");
});

test("handleChange functionality", () => {
    let profileData={email:"testedValue@gmail.com" , phoneNumber:123456788};
    let errors={newPasswordErrors:"newPasswordErrors" , confirmErrors:"confirmErrors"};
  let  wrapper = mount(<EditUserProfileByAdmin profileData={profileData} errors={errors} />) ;
  expect(wrapper.find("#edit-profile-by-admin-email").at(0).props().value).toBe("testedValue@gmail.com");
  expect(wrapper.find("#edit-profile-by-admin-tel").at(0).props().value).toBe(123456788);
  expect(wrapper.find("#edit-profile-by-admin-outlined-adornment-password").at(0).props().error).toBeTruthy();
  expect(wrapper.find("#edit-profile-by-admin-outlined-adornment-password").at(0).props().type).toBe("password");
  expect(wrapper.find("#edit-profile-by-admin-outlined-adornment-password").at(0).props().value).toBe("");
  expect(wrapper.find(Visibility).length).toBe(0);
  expect(wrapper.find(VisibilityOff).length).toBe(1);
  expect(wrapper.find("#edit-profile-by-admin-outlined-adornment-password-helper-text").at(0).props().error).toBeTruthy();
  expect(wrapper.find("#edit-profile-by-admin-confirmPassword").at(0).props().error).toBeTruthy();
  expect(wrapper.find("#edit-profile-by-admin-confirmPassword").at(0).props().helperText).toBe("confirmErrors");

  act(() => {
    wrapper
      .find("#edit-profile-by-admin-outlined-adornment-password")
      .at(0)
      .props()
      .onChange({ target: { value: "newPassword"} });
  });
  wrapper.render();
    wrapper.update();

    expect(wrapper.find("#edit-profile-by-admin-email").at(0).props().value).toBe("testedValue@gmail.com");
    expect(wrapper.find("#edit-profile-by-admin-tel").at(0).props().value).toBe(123456788);
    expect(wrapper.find("#edit-profile-by-admin-outlined-adornment-password").at(0).props().error).toBeTruthy();
    expect(wrapper.find("#edit-profile-by-admin-outlined-adornment-password").at(0).props().type).toBe("password");
    expect(wrapper.find("#edit-profile-by-admin-outlined-adornment-password").at(0).props().value).toBe("newPassword");
    expect(wrapper.find(Visibility).length).toBe(0);
    expect(wrapper.find(VisibilityOff).length).toBe(1);
    expect(wrapper.find("#edit-profile-by-admin-outlined-adornment-password-helper-text").at(0).props().error).toBeTruthy();
    expect(wrapper.find("#edit-profile-by-admin-confirmPassword").at(0).props().error).toBeTruthy();
    expect(wrapper.find("#edit-profile-by-admin-confirmPassword").at(0).props().helperText).toBe("confirmErrors");
});

test("handleClickShowPassword functionality", () => {
    let profileData={email:"testedValue@gmail.com" , phoneNumber:123456788};
    let errors={newPasswordErrors:"newPasswordErrors" , confirmErrors:"confirmErrors"};
  let  wrapper = mount(<EditUserProfileByAdmin profileData={profileData} errors={errors} />) ;
  expect(wrapper.find("#edit-profile-by-admin-email").at(0).props().value).toBe("testedValue@gmail.com");
  expect(wrapper.find("#edit-profile-by-admin-tel").at(0).props().value).toBe(123456788);
  expect(wrapper.find("#edit-profile-by-admin-outlined-adornment-password").at(0).props().error).toBeTruthy();
  expect(wrapper.find("#edit-profile-by-admin-outlined-adornment-password").at(0).props().type).toBe("password");
  expect(wrapper.find("#edit-profile-by-admin-outlined-adornment-password").at(0).props().value).toBe("");
  expect(wrapper.find(Visibility).length).toBe(0);
  expect(wrapper.find(VisibilityOff).length).toBe(1);
  expect(wrapper.find("#edit-profile-by-admin-outlined-adornment-password-helper-text").at(0).props().error).toBeTruthy();
  expect(wrapper.find("#edit-profile-by-admin-confirmPassword").at(0).props().error).toBeTruthy();
  expect(wrapper.find("#edit-profile-by-admin-confirmPassword").at(0).props().helperText).toBe("confirmErrors");

  act(() => {
    wrapper
      .find("#iconButton")
      .at(0)
      .props()
      .onClick();
  });
  wrapper.render();
    wrapper.update();

    expect(wrapper.find("#edit-profile-by-admin-email").at(0).props().value).toBe("testedValue@gmail.com");
    expect(wrapper.find("#edit-profile-by-admin-tel").at(0).props().value).toBe(123456788);
    expect(wrapper.find("#edit-profile-by-admin-outlined-adornment-password").at(0).props().error).toBeTruthy();
    expect(wrapper.find("#edit-profile-by-admin-outlined-adornment-password").at(0).props().type).toBe("text");
    expect(wrapper.find("#edit-profile-by-admin-outlined-adornment-password").at(0).props().value).toBe("");
    expect(wrapper.find(Visibility).length).toBe(1);
    expect(wrapper.find(VisibilityOff).length).toBe(0);
    expect(wrapper.find("#edit-profile-by-admin-outlined-adornment-password-helper-text").at(0).props().error).toBeTruthy();
    expect(wrapper.find("#edit-profile-by-admin-confirmPassword").at(0).props().error).toBeTruthy();
    expect(wrapper.find("#edit-profile-by-admin-confirmPassword").at(0).props().helperText).toBe("confirmErrors");
});

test("handleMouseDownPassword functionality", () => {
    let profileData={email:"testedValue@gmail.com" , phoneNumber:123456788};
    let errors={newPasswordErrors:"newPasswordErrors" , confirmErrors:"confirmErrors"};
    const preventDefaultMock = jest.fn();
  let  wrapper = mount(<EditUserProfileByAdmin profileData={profileData} errors={errors} />) ;
  expect(wrapper.find("#edit-profile-by-admin-email").at(0).props().value).toBe("testedValue@gmail.com");
  expect(wrapper.find("#edit-profile-by-admin-tel").at(0).props().value).toBe(123456788);
  expect(wrapper.find("#edit-profile-by-admin-outlined-adornment-password").at(0).props().error).toBeTruthy();
  expect(wrapper.find("#edit-profile-by-admin-outlined-adornment-password").at(0).props().type).toBe("password");
  expect(wrapper.find("#edit-profile-by-admin-outlined-adornment-password").at(0).props().value).toBe("");
  expect(wrapper.find(Visibility).length).toBe(0);
  expect(wrapper.find(VisibilityOff).length).toBe(1);
  expect(wrapper.find("#edit-profile-by-admin-outlined-adornment-password-helper-text").at(0).props().error).toBeTruthy();
  expect(wrapper.find("#edit-profile-by-admin-confirmPassword").at(0).props().error).toBeTruthy();
  expect(wrapper.find("#edit-profile-by-admin-confirmPassword").at(0).props().helperText).toBe("confirmErrors");

  act(() => {
    wrapper
      .find("#iconButton")
      .at(0)
      .props()
      .onMouseDown({ preventDefault: preventDefaultMock });
  });
  wrapper.render();
    wrapper.update();
    expect(preventDefaultMock).toHaveBeenCalledTimes(1);
    expect(wrapper.find("#edit-profile-by-admin-email").at(0).props().value).toBe("testedValue@gmail.com");
    expect(wrapper.find("#edit-profile-by-admin-tel").at(0).props().value).toBe(123456788);
    expect(wrapper.find("#edit-profile-by-admin-outlined-adornment-password").at(0).props().error).toBeTruthy();
    expect(wrapper.find("#edit-profile-by-admin-outlined-adornment-password").at(0).props().type).toBe("password");
    expect(wrapper.find("#edit-profile-by-admin-outlined-adornment-password").at(0).props().value).toBe("");
    expect(wrapper.find(Visibility).length).toBe(0);
    expect(wrapper.find(VisibilityOff).length).toBe(1);
    expect(wrapper.find("#edit-profile-by-admin-outlined-adornment-password-helper-text").at(0).props().error).toBeTruthy();
    expect(wrapper.find("#edit-profile-by-admin-confirmPassword").at(0).props().error).toBeTruthy();
    expect(wrapper.find("#edit-profile-by-admin-confirmPassword").at(0).props().helperText).toBe("confirmErrors");
});
});
   