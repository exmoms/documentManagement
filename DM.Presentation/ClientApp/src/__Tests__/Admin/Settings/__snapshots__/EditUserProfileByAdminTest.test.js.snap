// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditUserProfileByAdmin matches snapshot 1`] = `
<WithStyles(ForwardRef(Container))
  component="main"
  maxWidth="xs"
>
  <WithStyles(CssBaseline) />
  <div
    className="makeStyles-paper-1"
  >
    <WithStyles(ForwardRef(Grid))
      container={true}
      spacing={1}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={12}
      >
        <WithStyles(ForwardRef(TextField))
          autoComplete="email"
          fullWidth={true}
          id="edit-profile-by-admin-email"
          label="email"
          name="edit-profile-by-admin-email"
          onChange={[Function]}
          required={true}
          type="email"
          value=""
          variant="outlined"
        />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={12}
      >
        <WithStyles(ForwardRef(TextField))
          autoComplete="tel"
          fullWidth={true}
          id="edit-profile-by-admin-tel"
          label="phone_number"
          name="edit-profile-by-admin-tel"
          onChange={[Function]}
          type="tel"
          value=""
          variant="outlined"
        />
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(FormControl))
        className="makeStyles-margin-4 makeStyles-textField-5"
        variant="outlined"
      >
        <WithStyles(ForwardRef(InputLabel))
          error={true}
          htmlFor="edit-profile-by-admin-outlined-adornment-password"
        >
          new_password
        </WithStyles(ForwardRef(InputLabel))>
        <WithStyles(ForwardRef(OutlinedInput))
          endAdornment={
            <ForwardRef(WithStyles)
              position="end"
            >
              <ForwardRef(WithStyles)
                aria-label="toggle password visibility"
                edge="end"
                id="iconButton"
                onClick={[Function]}
                onMouseDown={[Function]}
              >
                <UNDEFINED />
              </ForwardRef(WithStyles)>
            </ForwardRef(WithStyles)>
          }
          error={true}
          id="edit-profile-by-admin-outlined-adornment-password"
          labelWidth={120}
          onChange={[Function]}
          type="password"
          value=""
        />
        <WithStyles(ForwardRef(FormHelperText))
          error={true}
          id="edit-profile-by-admin-outlined-adornment-password-helper-text"
        >
          newPasswordErrors
        </WithStyles(ForwardRef(FormHelperText))>
      </WithStyles(ForwardRef(FormControl))>
      <WithStyles(ForwardRef(Grid))
        item={true}
        xs={12}
      >
        <WithStyles(ForwardRef(TextField))
          autoComplete="current-password"
          error={true}
          fullWidth={true}
          helperText="confirmErrors"
          id="edit-profile-by-admin-confirmPassword"
          label="confirmNewPassword_label"
          name="edit-profile-by-admin-confirmPassword"
          type="password"
          variant="outlined"
        />
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </div>
</WithStyles(ForwardRef(Container))>
`;
